syntax = "proto3";

service Raft {
    rpc RequestVote (RequestVoteRequest) returns (RequestVoteResponse) {}
    rpc AppendEntries (AppendEntriesRequest) returns (AppendEntriesResponse) {}
}

message RequestVoteRequest {
    uint32 term = 1;
    uint32 candidateId = 2;
    uint32 LastLogIndex = 3;
    uint32 LastLogTerm = 4;
}

message RequestVoteResponse {
    uint32 term = 1;
    bool voteGranted = 2;
}

message AppendEntriesRequest {
    uint32 term = 1;
    uint32 leaderId = 2;
    uint32 prevLogIndex = 3;
    uint32 prevLogTerm = 4;
    uint32 leaderCommit = 5;
}

message AppendEntriesResponse {
    uint32 term = 1;
    bool success = 2;
}
